### ISSUE TITLE
Training the model with new dataset

### ISSUE THREAD

**Issue by taxuezcy** [yysijie/st-gcn#64]
@yysijie @yjxiong ,hello, sorry to interrupt you, i have built my own dataset which is much smaller than kinetics, when i use the new dataset to train the model and adjust some parameters, the best accuracy is 47.77% for Top1, 90.23% for top5, but it is still not enough,could you help me to improve the accuracy?Looking forward to your reply!

**Comment by yysijie** [comment]
Hi, if you use a smaller dataset, your model may suffer from over-fitting problem. 
May be you can use our model as a pre-trained model.
Or, you can use some proper augmentation strategies, such as rotation, transform, randomly clipping and so on.

**Comment by taxuezcy** [comment]
@yysijie   hello, thank you very much for your kind reply! I have tried the method you proposed,but the accurary is still small. I wondor if it is the limit of the 2d pose, so i want to test the ntu-rgb dataset，but i cannot get the dataset,  i have send the request for ntu-rgb,but i don't receive the reply. so can you share me the ntu-rgb dataset, looking forward to your reply

**Comment by yysijie** [comment]
I think your problem is not caused by the limit of 2d pose. What is the accuracy?

Sorry, I can not share the ntu-rgb dataset without their authorization. They replied me on the third day.

**Comment by taxuezcy** [comment]
@yysijie ,thank you for your kind reply! i have used the PKUMMD dataset to test your 3d ntu-crossview pretained model，and the accuracy is only 3.52% for top1, i wonder what cause such a big difference?

**Comment by yysijie** [comment]
It's strange. Did you check the value range of data and the definition of joints are identical?

**Comment by ml930310** [comment]
@taxuezcy Hello, how do you generate a val_json file when you are doing a dataset?

**Comment by wwwpbai** [comment]
@taxuezcy , Hello , how do you generate train_json file when you use new dataset?Download dataset from the internet or generate by openpose?

**Comment by taxuezcy** [comment]
@fmthoker ,yes, I have trained st-gcn on PKUMMD dataset

**Comment by taxuezcy** [comment]
@ml930310 @wwwpbai ,I generate train_json file and  val_json file  by openpose

**Comment by wwwpbai** [comment]
@taxuezcy, use this commad to generate json file ? " ./build/examples/openpose/openpose.bin --video examples/media/video.avi"

**Comment by 645704621** [comment]
@taxuezcy @yysijie Hello，how to use the pre-trained model to train my dataset

**Comment by 1209805200** [comment]
> @ ml930310 @wwwpbai，我通过openpose生成train_json文件和val_json文件

i can't generate a file but some files of every frames. How can i generate a whole file？

**Comment by 1209805200** [comment]
> @ml930310 @wwwpbai ,I generate train_json file and val_json file by openpose

i can't generate a file but some files of every frames. How can i generate a whole file？

**Comment by wwwpbai** [comment]
@1209805200, you can use code [https://github.com/yysijie/st-gcn/blob/master/tools/kinetics_gendata.py](here) to generate a whole file with json files of every frame.

**Comment by anjingxing** [comment]
@taxuezcy  hi,sorry to interrupt you, i have built ucf101 dataset ,can i tell you what your order is ?How to transfer the original video dataset to the skeleton format file required by the network

**Comment by hacker-wei** [comment]
@wwwpbai soory to interrupt you, how to  get this red line file? i don't know how to run openpose to get my dataset json file.
![图片](https://user-images.githubusercontent.com/32689594/58677433-7b601580-838e-11e9-93e5-c956ced5d3d9.png)

**Comment by wwwpbai** [comment]
@hacker-wei, you can see [demo.py](https://github.com/yysijie/st-gcn/blob/master/processor/demo.py),this file tests some video files by calling openpose, you can learn from it then write your own code ,then run your own code get kinetics-skeleton then save them in your computer.

**Comment by hacker-wei** [comment]
@wwwpbai thank you. i'll try it. and how to get this file,do you have the script.py
???![图片](https://user-images.githubusercontent.com/32689594/58679039-6ab29e00-8394-11e9-9580-d5e33a29cace.png)

**Comment by wwwpbai** [comment]
@hacker-wei , you can see this [script](https://github.com/yysijie/st-gcn/blob/master/tools/kinetics_gendata.py)

**Comment by hacker-wei** [comment]
@wwwpbai thank you.may i have your wechat or qq?

**Comment by hacker-wei** [comment]
why this the output channel multiple kernel_size of conv operate?
![图片](https://user-images.githubusercontent.com/32689594/58772906-a98f6080-85ed-11e9-97f9-fe7b1f267566.png)

**Comment by hacker-wei** [comment]
can you give the url of openpose. build file in ubuntu to me? i can't compile the openpose, it waste my lots of time.thank you @yysijie

**Comment by i2isaalien** [comment]
> @ml930310 @wwwpbai ,I generate train_json file and val_json file by openpose
@taxuezcy  Hi,there is a problem coming to me.Do you get the train_json of your own dataset by openpose directly? (e.g. you don't add your custom post-processing function with the OpenPose C++/Python API.)

**Comment by i2isaalien** [comment]
> @ml930310 @wwwpbai ,I generate train_json file and val_json file by openpose

@taxuezcy  Hi,there is a problem coming to me.Do you get the train_json of your own dataset by the openpose directly?(e.g. you didn't add your custom post-processing function with the OpenPose C++/Python API ?)
thx!

**Comment by taxuezcy** [comment]
@i2isaalien，it needs two step to get the train_json

**Comment by i2isaalien** [comment]
> @i2isaalien，it needs two step to get the train_json

thanks to reply my question,Could you be more specific?

**Comment by YZYLHR** [comment]
> 为什么这个输出通道的conv的多个kernel_size操作？
> ![图片](https://user-images.githubusercontent.com/32689594/58772906-a98f6080-85ed-11e9-97f9-fe7b1f267566.png)

Do you know why?

**Comment by Akhilreddy7777** [comment]
Hello,
    I have .json files(skeleton data) for each frame. I want to get a .npy file out of it which the training script accepts. What script should I use ? I am thinking to use kinetics_gendata.py Is that correct?

Thanks in advance

**Comment by YZYLHR** [comment]
yes！You can use openpose. py to gendata. json file for each video. Then you can use&nbsp;kinetics_gendata.py to get the. npy file.&nbsp;



---Original---

**Comment by luoluo-gif** [comment]
**I used the self built data set to train the new model, but there was a situation of Top1 = 40%, top5 = 100%. Why?** 
**And I don't use openpost to extract bone data. I use joint point data extracted from other models. The effect of joint point position is not much worse than that of openpost, but there is a little difference in confidence. Does that have an impact?**

 @yysijie @taxuezcy @fmthoker @ml930310 @wwwpbai

**Comment by YZYLHR** [comment]
Hi! Maybe your dataset is too small.You can try using the pre-training model.






------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"luoluo-gif"<notifications@github.com&gt;;
发送时间:&nbsp;2020年1月3日(星期五) 上午10:02
收件人:&nbsp;"open-mmlab/mmskeleton"<mmskeleton@noreply.github.com&gt;;
抄送:&nbsp;"小仙女"<1161360870@qq.com&gt;;"Comment"<comment@noreply.github.com&gt;;
主题:&nbsp;Re: [open-mmlab/mmskeleton] Training the model with new dataset (#64)




I used the self built data set to train the new model, but there was a situation of Top1 = 40%, top5 = 100%. Why?
 And I don't use openpost to extract bone data. I use joint point data extracted from other models. The effect of joint point position is not much worse than that of openpost, but there is a little difference in confidence. Does that have an impact?
 
@yysijie @taxuezcy @fmthoker @ml930310 @wwwpbai

**Comment by Summer-sss** [comment]
> I think your problem is not caused by the limit of 2d pose. What is the accuracy?
> 
> Sorry, I can not share the ntu-rgb dataset without their authorization. They replied me on the third 
你好，请问Kinetics skeleton 数据集方便分享以下百度网盘的连接吗，连接失效了。

**Comment by Jagadish-Kumaran** [comment]
> @yysijie ,thank you for your kind reply! i have used the PKUMMD dataset to test your 3d ntu-crossview pretained model，and the accuracy is only 3.52% for top1, i wonder what cause such a big difference?

Were you able resolve this issue?

**Comment by zjr666** [comment]
> 
> yes！You can use openpose. py to gendata. json file for each video. Then you can use&nbsp;kinetics_gendata.py to get the. npy file.&nbsp;
> […](#)
> ---Original--- From: "Akhil"<notifications@github.com&gt; Date: Wed, Nov 13, 2019 08:18 AM To: "open-mmlab/mmskeleton"<mmskeleton@noreply.github.com&gt;; Cc: "YZYLHR"<1161360870@qq.com&gt;;"Comment"<comment@noreply.github.com&gt;; Subject: Re: [open-mmlab/mmskeleton] Training the model with new dataset (#64) Hello, I have .json files(skeleton data) for each frame. I want to get a .npy file out of it which the training script accepts. What script should I use ? I am thinking to use kinetics_gendata.py Is that correct? Thanks in advance — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.

What is the file format of data_path and label_path in kinetics_gendata.py?

**Comment by 15717097043** [comment]
> @fmthoker，是的，我已经在PKUMMD数据集上训练了st-gc
Is there a tutorial or post

**Comment by zren2** [comment]
> yes！You can use openpose. py to gendata. json file for each video. Then you can use&nbsp;kinetics_gendata.py to get the. npy file.&nbsp;
> […](#)

where is the openpose.py? I can't find it, Is it need write by myself?

**Comment by wyz0916-gitHub** [comment]
> @wwwpbai soory to interrupt you, how to get this red line file? i don't know how to run openpose to get my dataset json file. ![图片](https://user-images.githubusercontent.com/32689594/58677433-7b601580-838e-11e9-93e5-c956ced5d3d9.png)

Hi,how to install the openpose python-API.I have just contact it.
